app/page.jsx

```diff
@@ -62,6 +64,7 @@ const Dashboard = () => {
       </div>
 
       <div>
+        {tickets.length === 0 ? <NoTicketExists /> : ""}
```

Comment:
move the `getUniqueGroups()` call into the `else` block, so it only executes when tickets are present. The code should look like this:

```javascript
{tickets.length === 0 ? <NoTicketExists /> : 
getUniqueGroups().map((group, groupIndex) => (
  <div key={groupIndex} className="mb-4">
    <h2>{group}</h2>
  ....
```